tap_impact/sync.py

```diff
@@ -280,48 +287,56 @@ def sync_endpoint(client,
-                            child_bookmark_field = next(iter(child_endpoint_config.get(
-                                'replication_keys', [])), None)
-                            child_total_records = sync_endpoint(
-                                client=client,
-                                catalog=catalog,
-                                state=state,
-                                start_date=start_date,
-                                stream_name=child_stream_name,
-                                path=child_path,
-                                endpoint_config=child_endpoint_config,
-                                static_params=child_endpoint_config.get('params', {}),
-                                bookmark_query_field=child_endpoint_config.get(
-                                    'bookmark_query_field'),
-                                bookmark_field=child_bookmark_field,
-                                bookmark_type=child_endpoint_config.get('bookmark_type'),
-                                data_key=child_endpoint_config.get('data_key', 'results'),
-                                id_fields=child_endpoint_config.get('key_properties'),
-                                selected_streams=selected_streams,
-                                parent=child_endpoint_config.get('parent'),
-                                parent_id=parent_id)
-                            LOGGER.info(
-                                'FINISHED Sync for Stream: {}, parent_id: {}, total_records: {}'\
-                                    .format(child_stream_name, parent_id, child_total_records))
+                        model_id = config.get('model_id')
+                        process_child = True
+                        # conversion_paths endpoint requires model_id tap config param
+                        if child_stream_name == 'conversion_paths' and not(model_id):
+                            process_child = False
+                        if process_child: 
+                            write_schema(catalog, child_stream_name)
+                            # For each parent record
+                            for record in transformed_data:
+                                i = 0
+                                # Set parent_id
+                                for id_field in id_fields:
+                                    if i == 0:
+                                        parent_id_field = id_field
+                                    if id_field == 'id':
+                                        parent_id_field = id_field
+                                    i = i + 1
+                                parent_id = record.get(parent_id_field)
+
+                                # sync_endpoint for child
+                                LOGGER.info(
+                                    'START Sync for Stream: {}, parent_stream: {}, parent_id: {}'\
+                                        .format(child_stream_name, stream_name, parent_id))
```

Comment:
We're going to run `pylint` on this code eventually (if we aren't already), and this will fail. Please make it `LOGGER.info('START Sync for Stream: %s, parent_stream: %s, parent_id: %s', child_stream_name, stream_name, parent_id)`, which will also fix your line continuation problem

Reply:
.format vs. %s - please see the first response here:
https://stackoverflow.com/questions/5082452/string-formatting-vs-format

I prefer .format for those reasons. It does not return a pylint error. I think it is easier to use and does all of the datatype conversions to put into the string for the LOGGER without any issues - despite the source datatype.

Reply:
Also see old way vs. new way: https://pyformat.info/#simple

Reply:
`pylint` errors with `W1202: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)`

Reply:
Strange - I don't get a pylint error. Maybe you are using an old version of pylint.